zerver/middleware.py

```diff
@@ -452,7 +452,7 @@ def process_exception(
         self, request: HttpRequest, exception: Exception
     ) -> Optional[HttpResponse]:
         if isinstance(exception, MissingAuthenticationError):
-            if "text/html" in request.META.get("HTTP_ACCEPT", ""):
+            if "text/html" in request.headers.get("Accept", ""):
```

Comment:
Should we be using [`request.accepts("text/html")`](https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpRequest.accepts) ?

Reply:
I considered that at first, but it fails lots of tests because it’s too correct—it treats the default as `*/*` and matches that against `text/html`.

This entire block doesn’t make much sense, as its comment admits. Redirecting /api/v1 to the login page doesn’t even help you authenticate to /api/v1, and /json isn’t a public interface.

Reply:
OK, makes sense!